### PR TITLE
chore: regenerate knip reports

### PR DESCRIPTION
**Pull request by 04kash**
## Hey, I just made a Pull Request!

- Regenerate knip reports for all workspaces
- Make knip a devDependency instead of a dependency

Fixes: https://issues.redhat.com/browse/RHIDP-4959

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/redhat-developer/rhdh-plugins/blob/main/CONTRIBUTING.md#creating-changesets))
- [ ] Added or Updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

File: workspaces/dynamic-home-page/.yarn/install-state.gz
Comment (christoph-jerolimov) [inline workspaces/dynamic-home-page/.yarn/install-state.gz:1]: Can you remove this file  from your PR, its a left over after I renamed the `dynamic-home-page` workspace to `homepage`.

File: workspaces/bulk-import/packages/app/knip-report.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Knip report
+
+## Unused dependencies (5)
+
+| Name                                                | Location     | Severity |
+| :-------------------------------------------------- | :----------- | :------- |
+| @red-hat-developer-hub/backstage-plugin-bulk-import | package.json | error    |
```
Comment (christoph-jerolimov) [inline workspaces/bulk-import/packages/app/knip-report.md:7]: Do you understand this false-positive?

The import is used in `workspaces/bulk-import/packages/app/src/App.tsx`

File: workspaces/bulk-import/packages/backend/knip-report.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Knip report
+
+## Unused dependencies (11)
+
+| Name                                                        | Location     | Severity |
+| :---------------------------------------------------------- | :----------- | :------- |
+| @red-hat-developer-hub/backstage-plugin-bulk-import-backend | package.json | error    |
```
Comment (christoph-jerolimov) [inline workspaces/bulk-import/packages/backend/knip-report.md:7]: Same here, it's used... Hmm :thinking:

**Review by christoph-jerolimov — CHANGES_REQUESTED** [review]
Hi, please remove at least the yarn blob (you can remove th dynamic-home-page workspace from your machine).

Also I see some false positives for "unused dependencies". I don't know how useful this is when the report is incorrect.

We can merge anyway but I think we should also check that we check that the file is accurate  (generate and ensure there is no diff)  on each PR.

File: workspaces/dynamic-home-page/.yarn/install-state.gz
Comment (04kash) [reply workspaces/dynamic-home-page/.yarn/install-state.gz:1]: Totally missed that, thanks @christoph-jerolimov !

**Review by christoph-jerolimov — APPROVED** [review]
Feel free to merge it when you're fine with it.

With two things we should follow-up:

1. Check/remove the false positives
2. Add this as a PR check (the idea is similar to the api-report.md, it makes changes like unused dependencies easier to notice)

**Comment by 04kash** [comment]
> Add this as a PR check (the idea is similar to the api-report.md, it makes changes like unused dependencies easier to notice)
I was actually thinking of setting up a workflow to run this script as a cron job on a regular basis: https://github.com/redhat-developer/rhdh-plugins/blob/main/scripts/ci/generate-knip-report.js

But your suggestion to include it in the PR checks also sounds good

**Comment by 04kash** [comment]
Merging this as is, will later open a follow-up PR to remove false positives.

**Pull request merged**